Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/XldbProtobufExtensions.cs

```diff
@@ -558,5 +548,479 @@ public static Xldb.Fingerprint ToFingerprint(this Fingerprint fingerprint)
+                IsValid = pip.ServiceInfo.IsValid,
+                IsStartOrShutdownKind = pip.ServiceInfo.IsStartOrShutdownKind
+            };
+
+            serviceInfo.ServicePipDependencies.AddRange(pip.ServiceInfo.ServicePipDependencies.Select(key => key.Value));
+            serviceInfo.FinalizationPipIds.AddRange(pip.ServiceInfo.FinalizationPipIds.Select(key => key.Value));
+
+            xldbProcessPip.ServiceInfo = serviceInfo;
+            xldbProcessPip.EnvironmentVariable.AddRange(pip.EnvironmentVariables.Select(
+                envVar => new EnvironmentVariable() { Name = envVar.Name.ToString(), Value = envVar.Value.ToPipData(), IsPassThrough = envVar.IsPassThrough }));
+            xldbProcessPip.Dependencies.AddRange(pip.Dependencies.Select(file => file.ToFileArtifact(pathTable)));
+            xldbProcessPip.DirectoryDependencies.AddRange(pip.DirectoryDependencies.Select(dir => dir.ToDirectoryArtifact(pathTable)));
+            xldbProcessPip.OrderDependencies.AddRange(pip.OrderDependencies.Select(order => order.Value));
+            xldbProcessPip.UntrackedPaths.AddRange(pip.UntrackedPaths.Select(path => path.ToAbsolutePath(pathTable)));
+            xldbProcessPip.UntrackedScopes.AddRange(pip.UntrackedScopes.Select(path => path.ToAbsolutePath(pathTable)));
+            xldbProcessPip.SuccessExitCodes.AddRange(pip.SuccessExitCodes.Select(code => code));
+            xldbProcessPip.RetryExitCodes.AddRange(pip.RetryExitCodes.Select(code => code));
+            xldbProcessPip.FileOutputs.AddRange(pip.FileOutputs.Select(
+                output => new Xldb.FileArtifactWithAttributes()
+                { IsValid = output.IsValid, Path = output.Path.ToAbsolutePath(pathTable), RewriteCount = output.RewriteCount, FileExistence = (Xldb.FileExistence)output.FileExistence }));
+            xldbProcessPip.DirectoryOutputs.AddRange(pip.DirectoryOutputs.Select(dir => dir.ToDirectoryArtifact(pathTable)));
+            xldbProcessPip.Semaphores.AddRange(pip.Semaphores.Select(
+                semaphore => new ProcessSemaphoreInfo() { Name = semaphore.Name.ToString(), Value = semaphore.Value, Limit = semaphore.Limit, IsValid = semaphore.IsValid }));
+            xldbProcessPip.AdditionalTempDirectories.AddRange(pip.AdditionalTempDirectories.Select(dir => dir.ToAbsolutePath(pathTable)));
+            xldbProcessPip.PreserveOutputWhitelist.AddRange(pip.PreserveOutputWhitelist.Select(path => path.ToAbsolutePath(pathTable)));
+            xldbProcessPip.ServicePipDependencies.AddRange(pip.ServicePipDependencies.Select(dep => dep.Value));
+
+            if (pip.Tags.IsValid)
+            {
+                xldbProcessPip.Tags.AddRange(pip.Tags.Select(key => key.ToString()));
+            }
+
+            return xldbProcessPip;
+        }
+
+        /// <nodoc />
+        public static Xldb.HashSourceFile ToHashSourceFile(this HashSourceFile pip, PathTable pathTable, Xldb.Pip parentPip)

```

Comment:
>HashSourceFile [](start = 27, length = 14)

I'm not sure it's worth exposing HSF pips in the graph

Reply:
Are they not used anywhere else at all (any analyzers) or important at all? I put them in there since they are relatively lightweight and could have the potential of being used?(but maybe not)

---
In reply to: [314448106](https://github.com/microsoft/BuildXL/pull/724#discussion_r314448106) [](ancestors = 314448106)